### DISCUSSION TITLE
Entries in Test Navigator not removed

### DISCUSSION THREAD

**Discussion by alexvoss** [Tyler-Keith-Thompson/CucumberSwift#43] [Q&A]
Hi,

I am editing my feature files quite heavily to ensure the syntax is correct, that I remove duplicate step definitions and get the language right. When I run the tests the number of steps listed in the test navigator keeps rising. I can clear the list by restarting Xcode. Is there another way? Would it be possible to automate weeding out steps that no longer exist? Am I missing something?

Cheers,

Alex

**Comment by Tyler-Keith-Thompson** [comment]
Unfortunately nothing that I'm aware of. Apple doesn't really make it easy on people trying to integrate with XCTest. There may be some private method I could call or swizzle, but I'd have to be aware of it. 

That said in Xcode 13 I have found it clears if I hit `⌘ + K` to clean the target. It should be less cumbersome than restarting. There's also a report spit out by CucumberSwift you could use, but thanks to derived data it can be tricky to find. You'd also need something that can parse Cucumber JSON and spit out a rendered report. 

Lastly, you could view your log in the report navigator rather than looking through the test navigator. 

I'll do a little digging to see if anybody else has some magic way of clearing it that I've just never been aware of.

**Reply by alexvoss** [reply]
You mean `⌘ + ⌥ + K` == `Product/Clean Test Results? This does not quite do what I wanted as it removes test failure notifications from the code but not generated step names from the test navigator. I thought I had a situation where with every run the number of tests was increasing but I am unable to reproduce this now. Thanks for the help, as always much appreciated. I'll be in touch if I find the problem again and can reproduce it.

**Reply by Tyler-Keith-Thompson** [reply]
Yeah, the problem you witnessed is real, as you change scenario names you will end up with old tests left over in the navigator. I believe that cleaning and re-running ends up removing those leftover tests. 

What you should not get is an ever-expanding set of tests. In other words, if it has the same name, it will not continually add new entries. Only if the name changes.
